Fitbit.Portable/FitbitClient.cs

```diff
@@ -964,6 +908,164 @@ public Task<TimeSeriesDataListInt> GetTimeSeriesIntAsync(TimeSeriesResourceType
+            await HandleResponse(response);
+            var responseBody = await response.Content.ReadAsStringAsync();
+            return (new JsonDotNetSerializer() { RootProperty = "activityLog" }).Deserialize<ActivityLog>(responseBody);
+        }
+
+        /// <summary>
+        /// The Get Activity Logs List endpoint retrieves a list of a user's activity log entries before or after a given day with offset and limit using units in the unit system which corresponds to the Accept-Language header provided. - https://dev.fitbit.com/docs/activity/#get-activity-logs-list
+        /// </summary>
+        /// <param name="beforeDate">The date in the format yyyy-MM-ddTHH:mm:ss. Only yyyy-MM-dd is required. Either beforeDate or afterDate must be specified. Set sort to desc when using beforeDate.</param>
+        /// <param name="afterDate">The date in the format yyyy-MM-ddTHH:mm:ss. Only yyyy-MM-dd is required. Either beforeDate or afterDate must be specified. Set sort to asc when using afterDate.</param>
+        /// <param name="limit">The max of the number of entries returned (maximum: 20)</param>
+        /// <param name="encodedUserId">encoded user id, can be null for current logged in user</param>
+        /// <returns>ActivityLog</returns>
+        public async Task<List<ActivityLogsList>> GetActivityLogsList(DateTime? beforeDate, DateTime? afterDate, int limit = 20, string encodedUserId = default(string))
+        {
+            var apiCall = string.Empty;
+            limit = limit > 20 ? 20 : limit;
+            const int offset = 0;
+            var sort = string.Empty;
+            var dateString = string.Empty;
+            var date = string.Empty;
+
+            if (beforeDate != null && afterDate != null)
+            {
+                throw new ArgumentException("Please only specify a beforeDate or afterDate, not both: https://dev.fitbit.com/docs/activity/#get-activity-logs-list");
+            }
+
+            if (beforeDate != null)
+            {
+                dateString = "beforeDate";
+                date = beforeDate.Value.ToString("yyyy-MM-ddTHH:mm:ss");
+                sort = "desc";
+            }
+            if (afterDate != null)
+            {
+                dateString = "afterDate";
+                date = afterDate.Value.ToString("yyyy-MM-ddTHH:mm:ss");
+                sort = "asc";
+            }
+
+            apiCall = FitbitClientHelperExtensions.ToFullUrl("/1/user/{0}/activities/list.json?{1}={2}&sort={3}&limit={4}&offset={5}", encodedUserId, dateString, date, sort, limit, offset);
+
+            HttpResponseMessage response = await HttpClient.GetAsync(apiCall);
+            await HandleResponse(response);
+            var responseBody = await response.Content.ReadAsStringAsync();
+            return (new JsonDotNetSerializer() { RootProperty = "activities" }).Deserialize<List<ActivityLogsList>>(responseBody);
+        }
+
+

```

Comment:
Extra whitespace not needed between methods

Reply:
👍